tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/buildorder/BuildOrderTest.xtend

```diff
@@ -38,18 +43,36 @@ class BuildOrderTest extends AbstractIdeTest {
 	 *            test workspace
 	 */
 	@SafeVarargs
-	 def final void test(String expectation, Pair<String, List<Pair<String, String>>>... projectsModulesContents) {
+	 def final void test(String buildOrder, Pair<String, List<Pair<String, String>>>... projectsModulesContents) {
+		test(buildOrder, emptyList, projectsModulesContents);
+	}
+
+	/**
+	 * @param expectation
+	 *            string of project names ordered in build order
+	 * @param projectsModulesContents
+	 *            test workspace
+	 */
+	@SafeVarargs
+	 def final void test(String buildOrder, Collection<Collection<String>> cycles, Pair<String, List<Pair<String, String>>>... projectsModulesContents) {
 		testWorkspaceManager.createTestOnDisk(projectsModulesContents);
 		startAndWaitForLspServer();
 
-		val projectBuildOrderInfo = projectBuildOrderInfoProvider.get();
+		val workspaceConfig = workspaceConfigProvider.get();
+		val projectBuildOrderInfo = projectBuildOrderInfoProvider.get(workspaceConfig);
 		val boIterator = projectBuildOrderInfo.getIterator().visitAll();
 		try {
 			val String names = IteratorExtensions.join(boIterator, ", ", [it.name]);
-			assertEquals(expectation, names);
+			assertEquals(buildOrder, names);
 		} catch (Exception exc) {
 			throw new RuntimeException("Never happens since toString never throws an exception. Bogus xtext warning", exc)
 		};
+		assertEquals(cycles.size, projectBuildOrderInfo.projectCycles.size);
+		val expectedCycles = cycles.map[Strings.join(", ", it)].toSet;
+		for (cycle : projectBuildOrderInfo.projectCycles) {
+			val detectedCycle = Strings.join(", ", cycle);
+			assertTrue(expectedCycles.contains(detectedCycle));
+		}
```

Comment:
This test seems to assert only the correct computation of build order.

What we also need for the validation are a couple of tests that make sure that the cycle-related error(s) show up and disappear correctly during certain editing scenarios. Since you are no longer using `#reportProjectIssue()` it should work fairly reliably, so we don't need too many of those, but 1-3 or so would be good to have.